### PR TITLE
Split doxygen pages from spec and TN into files

### PR DESCRIPTION
**Pull request by byrnHDF**
Improves maintenance of pages involved.

### CODE REVIEW

File: doxygen/dox/AppDbgging.dox [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+display the file and line number within the library and the invariant condition that
+failed. A core dump may be generated for post mortem debugging. The code to perform these
+checks can be included on a per-package bases.
+
+\subsection subsec_adddbg_intro_stats Timings and Statistics
+The library can be configured to accumulate certain statistics about things like cache
+performance, datatype conversion, data space conversion, and data filters. The code is
+included on a per-package basis and enabled at runtime by an environment variable.
+
+\subsection subsec_adddbg_intro_trace API Tracing
```
Comment (mattjala) [inline doxygen/dox/AppDbgging.dox:28]: API Tracing was removed in #4341

File: doxygen/dox/TableSpec.dox [hunk 2]
```diff
@@ -0,0 +1,100 @@
+
+/** \page TBLSPEC HDF5 Table Specification Version 1.0
+The HDF5 specification defines the standard objects and storage for the standard HDF5
+objects. (For information about the HDF5 library, model and specification, see the HDF
+documentation.) This document is an additional specification do define a standard profile
```
Comment (mattjala) [inline doxygen/dox/TableSpec.dox]: `do define` -> `to define`

File: doxygen/dox/Unicode.dox [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+When working with Unicode text, one can no longer assume a 1:1 correspondence between the
+number of characters and the data storage requirement.
+
+\subsection subsec_unicode_warn_sys System Dependencies
+<b>Linux</b>, <b>Unix</b>, and similar systems generally handle UTF-8 encodings in correct and
+predictable ways. There is an apparent consensus in the Linux community that "UTF-8 is just the right way to go."
+
+<b>Mac OS</b> systems generally handle UTF-8 encodings correctly.
+
+<b>Windows</b> systems use a different Unicode encoding, UCS-2 (discussed in this UTF-16 article) at
```
Comment (mattjala) [inline doxygen/dox/Unicode.dox]: No article linked here. Probably supposed to link to https://en.wikipedia.org/wiki/UTF-16

File: doxygen/dox/VFLTechNote.dox [hunk 4]
```diff
@@ -0,0 +1,1025 @@
...
+When a chunk of memory is freed the library adds it to a free list and allocation requests
+are satisfied from the free list before requesting memory from the file driver. Each type of
+allocation request enumerated above has its own free list, but the file driver can specify that
+certain object types can share a free list. It does so by providing an array which maps a
+request type to a free list. If any value of the map is H5MF_DEFAULT (zero) then the object's
+own free list is used. The special value H5MF_NOLIST indicates that the library should not
+attempt to maintain a free list for that particular object type, instead calling the file driver
+each time an object of that type is freed.
+
+Mappings predefined in the 'H5FDpublic.h' file are:
```
Comment (mattjala) [inline doxygen/dox/VFLTechNote.dox]: These are defined in `H5FDdevelop.h` at present

File: doxygen/dox/VFLTechNote.dox [hunk 5]
```diff
@@ -0,0 +1,1025 @@
...
+
+\section sec_vfl_data Data Functions
+These functions operate on data, transferring a region of the format address space between memory and files.
+
+\subsection subsec_vfl_data_cont Contiguous I/O Functions
+A driver must specify two functions to transfer data from the library to the file and vice versa.
+<table>
+<tr>
+<td><code>static herr_t read (H5FD_t *file, H5FD_mem_t type, hid_t dxpl, haddr_t addr, hsize_t size, void *buf)</code></td>
+<td>The read function reads data from file file beginning at address addr and continuing
```
Comment (mattjala) [inline doxygen/dox/VFLTechNote.dox]: `file file` -> `file`

**Review by mattjala — CHANGES_REQUESTED** [review]
A few typos and out of date info

**Comment by byrnHDF** [comment]
> A few typos and out of date info

Updated

**Review by mattjala — DISMISSED** [review]

**Review by lrknox — DISMISSED** [review]

**Review by lrknox — APPROVED** [review]

**Review by mattjala — APPROVED** [review]

**Pull request merged**